vagrant/Vagrantfile

```diff
@@ -0,0 +1,139 @@
+#  config.hostmanager.include_offline = true
+ 
+  config.landrush.enabled = true
+  config.landrush.tld = 'example.com'
+  config.landrush.guest_redirect_dns = false  
+
+  if ENV['DEPLOYMENT_TYPE'] == 'enterprise'
+    # vagrant-registration
+    if ENV.has_key?('SUB_USERNAME') && ENV.has_key?('SUB_PASSWORD')
+      config.registration.username = ENV['SUB_USERNAME']
+      config.registration.password = ENV['SUB_PASSWORD']
+    end
+    
+    # Proxy Information from environment
+    config.registration.proxy = PROXY = (ENV['PROXY'] || '')
+    config.registration.proxyUser = PROXY_USER = (ENV['PROXY_USER'] || '')
+    config.registration.proxyPassword = PROXY_PASSWORD = (ENV['PROXY_PASSWORD'] || '')
+    config.registration.auto_attach = false
+  end
+
+  config.vm.provider "virtualbox" do |v|
+     #v.customize ["setextradata", :id, "VBoxInternal2/SharedFoldersEnableSymlinksCreate//vagrant","1"]
+     v.memory = 1024
+     v.cpus = 1
+  end
+
+  config.vm.define :master1 do |master1|
+    master1.vm.box = box_name
+    #master1.vm.box_url = centos_box_url
+    master1.vm.network :private_network, ip: "#{NETWORK_BASE}.#{INTEGRATION_START_SEGMENT}"
+    master1.vm.hostname = "master1.example.com"
+    master1.vm.provision "shell", inline: "echo \"installing httpd-tools\""
+    master1.vm.provision "shell", inline: "sudo yum -y install httpd-tools"
+  end
+
+  config.vm.define :node1 do |node1|
+    node1.vm.box = box_name
+    #node1.vm.box_url = centos_box_url
+    node1.vm.network :private_network, ip: "#{NETWORK_BASE}.#{INTEGRATION_START_SEGMENT + 1}"
+    node1.vm.hostname = "node1.example.com"
+  end
+ 
+  config.vm.define :node2 do |node2|
+    node2.vm.box = box_name
+    #node2.vm.box_url = centos_box_url
+    node2.vm.network :private_network, ip: "#{NETWORK_BASE}.#{INTEGRATION_START_SEGMENT + 2}"
+    node2.vm.hostname = "node2.example.com"
+  end
+
+  config.vm.define :admin1 do |admin1|
+    admin1.vm.box = box_name
+    #admin1.vm.box_url = centos_box_url
+    admin1.vm.network :private_network, ip: "#{NETWORK_BASE}.#{INTEGRATION_START_SEGMENT + 3}"
+    admin1.vm.hostname = "admin1.example.com"
+
+    if ENV['DEPLOYMENT_TYPE'] == 'enterprise'
+      admin1.vm.provision "shell", path: "enterprise_installer.sh"
+    else
+      admin1.vm.provision "shell", path: "origin_installer.sh"
+    end 
+
+    admin1.vm.provision "shell", inline: "sudo cp -f /vagrant/etc_ansible_hosts /etc/ansible/hosts"
+    
+    if ENV['DEPLOYMENT_TYPE'] == 'enterprise'
+      admin1.vm.provision "shell", inline: "echo \"updating deployment type\""
+      admin1.vm.provision "shell", inline: "sudo cp /vagrant/update_deployment_type.sh /root"
+      admin1.vm.provision "shell", inline: "sudo /root/update_deployment_type.sh"
+    end
+
+  end
+
+  if ENV['DEPLOYMENT_TYPE'] == 'enterprise'
+    config.vm.provision "shell", inline: "echo \"setting up Red Hat subscriptions\""
+    config.vm.provision "shell", path: "repos.sh"
```

Comment:
I'm not sure there is a guarantee that this step is run prior to the admin1 provisioning.